zerver/views/push_notifications.py

```diff
@@ -101,3 +112,39 @@ def send_test_push_notification_api(
     send_test_push_notification(user_profile, devices)
 
     return json_success(request)
+
+
+@zulip_login_required
+def self_hosted_billing(request: HttpRequest) -> HttpResponse:
+    if not uses_notification_bouncer():
+        raise JsonableError(_("This server doesn't use the push notification bouncer."))
```

Comment:
Hmm, since this is a redirect to an HTML page, we probably don't want a JSON format error here? One could argue this should just return a standard Django 404 error, but probably we can resolve this the same way we address the `TODO` just below?

(Both errors should be basically unreachable)

Reply:
Yeah with the change in `gear_menu: Add proper logic for when to display remote billing`, this should be inaccessible in these cases - unless the user directly enters the URL in their browser. So 404 probably makes the most sense in the "not using bouncer" case - and in the other one, perhaps a json error is better than adding a new HTML template just for this? I added a commit with a comment like this:
```
        # We may want to replace this with an html error page at some point,
        # but this endpoint shouldn't be accessible via the UI to an unauthorized
        # user - and they need to directly enter the URL in their browser. So a json
        # error may be sufficient.
        raise OrganizationOwnerRequiredError
```

Reply:
Sounds good!